### DISCUSSION TITLE
VipsOperation: class "gifsave_buffer" not found

### DISCUSSION THREAD

**Discussion by elias-summermatter** [libvips/libvips#3161] [General]
**Describe the bug**
When i build with newest taged version (8.13.3) and try to resize gif images i get the following error:

```
undefined:0

Error: VipsOperation: class "gifsave_buffer" not found

Emitted 'error' event on Sharp instance at:
    at Object.<anonymous> (/wepublish/node_modules/sharp/lib/output.js:1218:16)
```

With the previous version (8.13.2) this does not happen.

**Environment**
(please complete the following information)
 - [Debian ](node:18.12.1-bullseye-slim)

**Comment by lovell** [comment]
Please provide full details of how you are compiling libvips, including which dependencies and versions you're building it with.

**Reply by elias-summermatter** [reply]
Used is docker image: 18.12.1-bullseye-slim
With debian packages: python2 gcc g++ make libjpeg62-turbo-dev libexif-dev liblcms2-dev fftw-dev libgif-dev libglib2.0-dev libpng-dev libwebp-dev libexpat-dev liborc-0.4-dev libtiff5-dev librsvg2-dev wget procps libjemalloc-dev libimagequant-dev

Build commands:
```
ARG LIBVIPS_VERSION=8.13.3
ARG LIBVIPS_SOURCE_TAR="vips-${LIBVIPS_VERSION}.tar.gz"
ARG LIBVIPS_SOURCE_URL="https://github.com/libvips/libvips/releases/download/v${LIBVIPS_VERSION}/${LIBVIPS_SOURCE_TAR}"
 mkdir ./libvips   && \
    cd ./libvips  && \
    wget ${LIBVIPS_SOURCE_URL} && \
    tar -xzf ${LIBVIPS_SOURCE_TAR} --strip 1   && \
    ./configure   --prefix=/usr   --enable-debug=no   --without-gsf   --disable-static   --mandir=/usr/share/man   --infodir=/usr/share/info   --docdir=/usr/share/doc   &&  \
    make && \
    make install && \
```

**Reply by jcupitt** [reply]
libvips has used cgif for GIF save since 8.12:

https://github.com/dloebl/cgif

You'll need to install that. You don't need libgif-dev any more.

**Reply by elias-summermatter** [reply]
Ok, then this is not possible to upgrade at the moment because debian stable (bullseye) does not include cgif, but will come available on sid. (https://packages.debian.org/unstable/libs/libcgif0). Its interessting that version 8.12.2 works but 8.12.3 doesn't. 
But still thanks for your fast answer :-)

**Reply by lovell** [reply]
It looks like you're using sharp; would its prebuilt binaries work for you? These provide all the dependencies listed here (and cgif). Perhaps you have a specific reason for installing a custom libvips?

**Reply by jcupitt** [reply]
Do you mean 8.13.3? I don't think there was a 8.12.3.

But I think they both work the same way -- without cgif, there's no dedicated GIF saver, but libvips will fall back to saving with imagemagick:

```
$ vips gifsave
vips: unknown action "gifsave"
$ vips copy 3198.gif x.gif
```

So without cgif `write_to_buffer` will work (though slowly), but `gifsave_buffer` will not.
